File: demo/src/main/java/com/backend/demo/entity/PickUp.java
package com.backend.demo.entity;

import java.time.LocalDate;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import com.backend.demo.serialization.CustomerSerializer;
import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;

@Entity
@Table(name="pickup")
@JsonIdentityInfo( generator = ObjectIdGenerators.PropertyGenerator.class,property = "id")

public class PickUp {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "id")
	private int id;
	
	@Column(name = "shipmentsCount")
	private int shipmentsCount;
	
	@Column(name="pickedShipments")
	private int pickedShipments;
	
	@Column(name="vehicleType")
	private String vehicleType;
	
	@Column(name="status")
	private String status;
	
	@Column(name="notes")
	private String notes;
	

	
	@ManyToOne
	@JoinColumn(name="customer_id")
    @JsonSerialize(using = CustomerSerializer.class)
	private Customer customer;
	
	@ManyToOne
	@JoinColumn(name="deliveryagent_id")
	private DeliveryAgent deliveryAgent;
	
	
	@Column
	private LocalDate date;
	
	public PickUp() {
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public int getShipmentsCount() {
		return shipmentsCount;
	}

	public void setShipmentsCount(int shipmentsCount) {
		this.shipmentsCount = shipmentsCount;
	}

	public int getPickedShipments() {
		return pickedShipments;
	}

	public void setPickedShipments(int pickedShipments) {
		this.pickedShipments = pickedShipments;
	}

	public String getVehicleType() {
		return vehicleType;
	}

	public void setVehicleType(String vehicleType) {
		this.vehicleType = vehicleType;
	}

	public String getStatus() {
		return status;
	}

	public void setStatus(String status) {
		this.status = status;
	}

	public String getNotes() {
		return notes;
	}

	public void setNotes(String notes) {
		this.notes = notes;
	}

	public Customer getCustomer() {
		return customer;
	}

	public void setCustomer(Customer customer) {
		this.customer = customer;
	}

	public LocalDate getDate() {
		return date;
	}

	public void setDate(LocalDate date) {
		this.date = date;
	}

	public DeliveryAgent getDeliveryAgent() {
		return deliveryAgent;
	}

	public void setDeliveryAgent(DeliveryAgent deliveryAgent) {
		this.deliveryAgent = deliveryAgent;
	}
	
	
	
}

File: mobile_app/lib/models/Pickup.dart
import 'package:mobile_app/models/Customer.dart';
import 'package:mobile_app/models/DeliveryAgent.dart';

class Pickup {
  final int? id;
  final int? shipmentsCount;
  final int? pickedShipments;
  final int? agentId;
  final int? customerId;
  final String? vehicleType;
  final String? status;
  final String? notes;
  final DateTime? date;
  final Customer? customer;
  final DeliveryAgent? deliveryAgent;

  Pickup({
    required this.id,
    required this.shipmentsCount,
    required this.vehicleType,
    this.pickedShipments,
    this.notes,
    this.deliveryAgent,
    this.customer,
    this.status,
    this.date,
    this.agentId,
    this.customerId,
  });

  factory Pickup.fromJson(json) {
    return Pickup(
      id: json['id'],
      shipmentsCount: json['shipmentsCount'],
      pickedShipments: json['pickedShipments'],
      vehicleType: json['vehicleType'],
      notes: json['notes'],
      status: json['status'],
      date: json['date'] != null
          ? DateTime.tryParse(json['date'])
          : DateTime.now(),
      // deliveryAgent: json['deliveryAgent'] != null
      //     ? DeliveryAgent.fromJson(json['deliveryAgent'])
      //     : null,
      // customer:
      //     json['customer'] != null ? Customer.fromJson(json['customer']) : null,
      agentId: json['agentId'],
      customerId: json['customerId'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'shipmentsCount': shipmentsCount,
      'pickedShipments': pickedShipments,
      'vehicleType': vehicleType,
      'notes': notes,
      'status': status,
      'date': date,
      'agentId': agentId,
      'customerId': customerId,
      // 'customer': customer?.toJson(),
      // 'deliveryAgent': deliveryAgent?.toJson()
    };
  }
}
